src/NHibernate/Cfg/Loquacious/IQueryCacheConfiguration.cs

```diff
@@ -1,9 +1,8 @@
-using NHibernate.Cache;
-
 namespace NHibernate.Cfg.Loquacious
 {
 	public interface IQueryCacheConfiguration
 	{
-		ICacheConfiguration Through<TFactory>() where TFactory : IQueryCacheFactory;
+		// 6.0 TODO: enable constraint
+		ICacheConfiguration Through<TFactory>(); // where TFactory : IQueryCacheFactory;
```

Comment:
Removing a constraint breaks compilation on implementing classes, at least for implicit implementations. I guess it is not a binary breaking change though, I do not think the IL cares for constraint.
This case is unfortunately not documented in the [breaking change guidelines](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md) of core fx.

Since this interface is not supposed to be implemented by anything outside of NHibernate code itself, I think it is acceptable in a minor release.